plugin/src/main/scala/sbtnativeimage/NativeImagePlugin.scala

```diff
@@ -110,23 +128,27 @@ object NativeImagePlugin extends AutoPlugin {
         out
       }
     },
-    nativeImageCommand := Def.taskDyn {
-      if (
+    nativeImageInstalled := Def.settingDyn {
+      val installed =
         "true".equalsIgnoreCase(System.getProperty("native-image-installed")) ||
-        "true".equalsIgnoreCase(System.getenv("NATIVE_IMAGE_INSTALLED"))
-      ) {
-        val binary =
-          if (Properties.isWin) "native-image.cmd" else "native-image"
-        val path =
-          Paths.get(System.getenv("JAVA_HOME")).resolve("bin").resolve(binary)
-        Def.task(List[String](path.toString()))
+        "true".equalsIgnoreCase(System.getenv("NATIVE_IMAGE_INSTALLED")) ||
+        "true".equalsIgnoreCase(System.getProperty("graalvm-installed")) ||
+        "true".equalsIgnoreCase(System.getenv("GRAALVM_INSTALLED"))
+      Def.setting(installed)
+    }.value,
+    nativeImageGraalHome := Def.taskDyn {
+      if (nativeImageInstalled.value) {
+        val path = Paths.get {
+          sys.env.get("GRAAL_HOME") orElse sys.env.get("GRAALVM_HOME") orElse sys.env.get("JAVA_HOME") getOrElse ""
```

Comment:
Nit: `sys.env` creates a new immutable copy of `System.getenv()` on every invocation.
```suggestion
          Option(System.getenv("GRAAL_HOME"))
            .orElse(Option(System.getenv("GRAALVM_HOME")))
            .orElse(Option(System.getenv("JAVA_HOME")))
            .getOrElse("")
```

Reply:
Alternatively,

```
List("GRAAL_HOME", "GRAALVM_HOME", "JAVA_HOME").iterator
  .map(key => Option(System.getenv(key)))
  .collectFirst { case Some(value) =>
    value
  }
```